### PR TITLE
Clean up docs for sim src, add LS-RTM examples

### PR DESCRIPTION
**Pull request by ziyiyin97**
1. Clean up the sim src documentation in the `doc` folder
2. Add LS-RTM example with anti-chattering
3. Add LS-RTM example with sim src

### CODE REVIEW

File: examples/scripts/lsrtm_example_antichatter.jl [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+F = judiModeling(info, model; options=opt)
+F0 = judiModeling(info, model0; options=opt)
+Ps = judiProjection(info, srcGeometry)
+J = judiJacobian(Pr*F0*adjoint(Ps), q)
+
+# Generate data via nonlinear modeling
+dobs = Pr*F*adjoint(Ps)*q
+dobs0 = Pr*F0*adjoint(Ps)*q
+
+# Get linearize data for LS-RTM
```
Comment (mloubout) [inline examples/scripts/lsrtm_example_antichatter.jl:104]: This is not linearized data

File: examples/scripts/lsrtm_example_antichatter.jl [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        global sumsign += sign.(g)
+        tau1 = t[j]*abs.(sumsign)/j
+        tau = t[j] * ones(Float32,size(C,1))
+        copyto!(view(tau, findall(flag_)), view(tau1, findall(flag_))) # use anti-chatter if pass the threshold
+
+        # update dual variable
+        global z -= t[j] * tau .* g
+    else
+        # update dual variable
+        global z -= t[j] * g
```
Comment (mloubout) [inline examples/scripts/lsrtm_example_antichatter.jl:171]: `@.`

File: examples/scripts/lsrtm_example_antichatter.jl [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    end
+
+    # estimate thresholding parameter in 1st iteration
+    (j==1) && (global lambda = quantile(abs.(z), .8))
```
Comment (mloubout) [inline examples/scripts/lsrtm_example_antichatter.jl:175]: Add sentence to explain what this is

File: examples/scripts/lsrtm_example_supershot.jl [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    J_j = judiJacobian(Pr_sim*F0_sim*adjoint(Ps_sim), q_j)
+
+    # superposition of shot records
+    data_j = Vector{Matrix{Float32}}(undef, batchsize)
+    for k = 1:batchsize
+        data_j[k] = sum(weights[k,:].*dlin.data)
+    end
+
+    # simultaneous shot record for current iteration
+    dlin_j = judiVector(sim_rec_geometry, data_j)
```
Comment (mloubout) [inline examples/scripts/lsrtm_example_supershot.jl:188]: That's a lot of array re-creation and stuff, a god utility function (or even better a linear operator) would be much nicer. I.e `W*R*d` with `W` weights and `R` resitriction

**Comment by ziyiyin97** [comment]
Thanks for your feedback. Running time is less than 1.5 minutes for each example on my laptop so I don't think that affects the CI run time that much. I'll clean things up a bit then

**Comment by ziyiyin97** [comment]
Ideal set-up for sim source is to have a JOLI operator `W` and it could be applied on `q`, `dobs`, `F`, `J`. Basically, I envisage the process of taking the gradient to look like

```julia
r = W * J * x - W * dobs
g = J' * W' * r
```

so that these symbols are exactly what they are in the paper.

I think doing `W*q` and `W*dobs` would be straightforward, but doing `W*F` and `W*J` needs some thought because we don't want to evaluate `F` or `J` first. Same deal for adjoint. My current thought is to do something like this

1. make `W` its own struct, namely `judiSimSrcW <: joAbstractLinearOperator`
2. overload `*(::judiSimSrcW, ::judiJacobian)` to output a new `judiJacobian`, NOT composition of JOLI operators
3. overload `*(::judiSimSrcW, ::judiPDEfull)` to output a new `judiPDEfull`

Any comment or suggestion? @mloubout thanks

**Comment by mloubout** [comment]
That's the idea yes. Need to be careful about weights redrawing and such as well

**Comment by mloubout** [comment]
FYI things gonna change (for better)
https://github.com/slimgroup/JUDI.jl/compare/refactor-22

So if you can keep it in mind. Only point source part is working for now.

**Comment by mloubout** [comment]
Closing as stale and ended up doing simsourcer myself since nothing was happening here

**Pull request closed without merging**